### PR TITLE
GR-848 add full-depth WGS page

### PR DESCRIPTION
**Pull request by callunity**

### CODE REVIEW

File: application/dash_application/utility/df_manipulation.py [hunk 1]
```diff
@@ -58,8 +111,51 @@
...
+_pinery_samples = _pinery_samples.drop(axis=1, columns=[PINERY_COL.NanodropConcentration, PINERY_COL.QubitConcentration, "RunIDandPosition", "TargetedResequencing", "TubeID", "RIN",
+"PrepKit", "PoolName", "Organism", "STRResult", "QubitConcentration", "SampleProvenanceID",
+"TissuePreparation", "TissueType", "SampleName", "RootSampleName", "Version", "NanodropConcentration",
+"TissueOrigin", "Purpose", "SequencerRunName", "SequencingParameters", "UMIs", "GroupIDDescription",
+"DV200", "CreateDate", "InstrumentName", "ParentSampleName", "SubProject", "TemplateType",
+"RunBaseMask", "RunDir", "ExternalName", "Institute", "StudyTitle", "LastModified",
+"GroupID", "SequencerRunPlatformModel", "ReceiveDate", "TissueRegion", "WorkflowType", "Skip",
+"LaneNumber", "LibrarySourceTemplateType"])
+
+unique_list = lambda vals: ", ".join(str(val) for val in sorted(set(vals)) if (val and val != "nan"))
```
Comment (slazicoicr) [inline application/dash_application/utility/df_manipulation.py:137]: Please comment the purpose of this (what will vals be?)

File: application/dash_application/utility/df_manipulation.py [hunk 2]
```diff
@@ -58,8 +111,51 @@
...
+    PINERY_COL.TissuePreparation: unique_list,
+    sample_type_col: unique_list,
+    # library attributes
+    PINERY_COL.PrepKit: unique_list,
+    PINERY_COL.TargetedResequencing: unique_list,
+    PINERY_COL.UMIs: unique_list,
+}
+    
+
+_pinery_merged_samples = _pinery_samples.fillna('').groupby(by=pinery_merged_columns).agg(concat_columns_for_merge).reset_index()
```
Comment (slazicoicr) [inline application/dash_application/utility/df_manipulation.py]: Please comment this line of code.

**Review by slazicoicr — COMMENTED** [review]
* BamQC Metrics data is missing from the table (Unmapped Reads % is empty)
* Highlight Samples does not work (I tried highlighting VAL1_0007)
* Cutoff boxes are named after the data value. I would either add change names of each box (PF Reads Tumour -> PF Reads Tumour Cutoff) or put a Cutoff header above all the boxes

**Comment by callunity** [comment]
> * BamQC Metrics data is missing from the table (Unmapped Reads % is empty)
> * Highlight Samples does not work (I tried highlighting VAL1_0007)
> * Cutoff boxes are named after the data value. I would either add change names of each box (PF Reads Tumour -> PF Reads Tumour Cutoff) or put a Cutoff header above all the boxes

Sorry, I explained about BamQC Metrics on the ticket but not in this PR. We don't yet have BamQC3 merged data, so that's why those graphs aren't being displayed. If anything, the unmapped reads graph should also have been commented out.

I'll address the other concerns.

For Cutoff boxes, I'm going to append each box label with "minimum" or "maximum", to make it clear which direction the cutoff applies to, unless there are any objections.

**Review by apmasell — APPROVED** [review]

File: application/dash_application/views/call_ready_wgs.py [hunk 3]
```diff
@@ -0,0 +1,549 @@
...
+from ..utility import df_manipulation as util
+from ..utility import sidebar_utils, log_utils
+
+logger = logging.getLogger(__name__)
+
+""" Set up elements needed for page """
+page_name = "call-ready-wgs"
+title = "Call-Ready WGS"
+
+ids = init_ids([
```
Comment (avarsava) [inline application/dash_application/views/call_ready_wgs.py:25]: Looks like some indentation weirdness happened here, but nbd

File: application/dash_application/views/call_ready_wgs.py [hunk 4]
```diff
@@ -0,0 +1,549 @@
...
+])
+
+BAMQC_COL = gsiqcetl.column.BamQc3MergedColumn
+ICHOR_COL = gsiqcetl.column.IchorCnaMergedColumn
+CALL_COL = gsiqcetl.column.MutetctCallabilityColumn
+PINERY_COL = pinery.column.SampleProvenanceColumn
+
+
+def dataversion():
+    return DATAVERSION
```
Comment (avarsava) [inline application/dash_application/views/call_ready_wgs.py:78]: Any reason we couldn't just use DATAVERSION wherever this is called?

**Review by avarsava — APPROVED** [review]

File: application/dash_application/utility/df_manipulation.py (hunk 1 above)
Comment (callunity) [reply application/dash_application/utility/df_manipulation.py:137]: Commented

File: application/dash_application/utility/df_manipulation.py (hunk 2 above)
Comment (callunity) [reply application/dash_application/utility/df_manipulation.py]: Commented

**Comment by callunity** [comment]
@slazicoicr I'd like to get this merged so we can build other reports off of it. I've made a note on the ticket that "highlight samples" still needs to be fixed, so that we don't lose that! Would that work for you?

File: application/dash_application/views/call_ready_wgs.py (hunk 4 above)
Comment (callunity) [reply application/dash_application/views/call_ready_wgs.py:78]: I'd have to test that out further to see if it's set when the known_pages_router requests it... perhaps in a subsequent PR?

File: application/dash_application/views/call_ready_wgs.py (hunk 3 above)
Comment (callunity) [reply application/dash_application/views/call_ready_wgs.py:25]: Ew. Fixed.

File: application/dash_application/views/call_ready_wgs.py (hunk 4 above)
Comment (avarsava) [reply application/dash_application/views/call_ready_wgs.py:78]: Sure!

**Review by slazicoicr — APPROVED** [review]

**Pull request merged**